### PR TITLE
fix: retain nsg rules during upgrade

### LINKED ISSUE #1624: Application not accessible from outside cluster during aks-engine Upgrade
Applications are not accessible from outside the cluster during the aks-engine upgrade

**Steps To Reproduce**
1. Deploy application(say nginx) and service of type LoadBalancer to expose the application to be accessed from outside the cluster
2. Upgrade cluster using aks-engine upgrade. During upgrade operation the application will not be accessible from outside the cluster

**Expected behavior**
Should be able to access the application from outside the cluster during upgrade process

**AKS Engine version**
v0.36.2

**Kubernetes version**
1.12.8

**Additional context**

### PR DESCRIPTION
**Pull request by Bhuvaneswari-Santharam**
**Reason for Change**:

Retain custom NSG rules created by workloads during upgrade scenario

**Issue Fixed**:

Fixes #1624

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
@Bhuvaneswari-Santharam is this PR based on top of current master?

**Comment by Bhuvaneswari-Santharam** [comment]
@jackfrancis Yes

File: pkg/engine/transform/transform.go [hunk 1]
```diff
@@ -454,6 +472,33 @@ func (t *Transformer) NormalizeResourcesForK8sMasterUpgrade(logger *logrus.Entry
 	return nil
 }
 
+//RemoveNsgDependency Removes the nsg dependency from the resource
+func RemoveNsgDependency(logger *logrus.Entry, resourceName string, resourceMap map[string]interface{}) {
+
+	if resourceName != "" && resourceMap != nil {
+		dependencies, ok := resourceMap[dependsOnFieldName].([]interface{})
+		if !ok {
+			logger.Warnf("Could not find dependencies for resourceName: %s", resourceName)
```
[the remark this answers is not in the document]
Comment (Bhuvaneswari-Santharam) [reply pkg/engine/transform/transform.go]: There may be resources without dependencies for which info should be good. Please let me know if this needs to be changed.

**Pull request merged**